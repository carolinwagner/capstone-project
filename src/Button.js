import React from 'react'
import styled from 'styled-components/macro'

export default function Button({ children, variant, ...rest }) {
  const buttonSwitch = {
    primary: <StyledPrimaryButton {...rest}>{children}</StyledPrimaryButton>,
    secondary: (
      <StyledSecondaryButton {...rest}>{children}</StyledSecondaryButton>
    ),
  }
  return buttonSwitch[variant]
}

const StyledBaseButton = styled.button`
  font-size: 1em;
  display: flex;
  justify-content: center;
  align-items: center;
  cursor: pointer;
  text-decoration: none;
  border-radius: 5px;
  border: none;
`

const StyledPrimaryButton = styled(StyledBaseButton)`
  padding: 20px;
  color: var(--lightgrey);
  background-color: var(--blue);
  box-shadow: 0 8px 16px 0 rgba(0, 0, 0, 0.2);
  flex: 1;
`

const StyledSecondaryButton = styled(StyledBaseButton)`
  padding: 10px;
  color: var(--lightgrey);
  background-color: var(--lightblue);
`
