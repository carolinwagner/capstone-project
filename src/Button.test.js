import React from 'react'
import '@testing-library/jest-dom/extend-expect'
import renderer from 'react-test-renderer'
import Button from './Button'

describe('Button', () => {
  it('renders Button correctly', () => {
    const tree = renderer.create(<Button />)
    expect(tree).toMatchSnapshot()
  })
})
