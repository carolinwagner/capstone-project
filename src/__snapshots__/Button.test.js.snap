// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders Button correctly 1`] = `
<button
  className="Button__StyledButton-sc-1p5kn3w-0 kAJRXX"
/>
`;
